server/controllers/payment.js

```diff
@@ -0,0 +1,77 @@
+const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
+const asyncHandler = require('express-async-handler');
+const User = require('../models/User');
+const Contest = require('../models/Contest');
+
+exports.setupUserForPayments = asyncHandler(async (req, res, next) => {
+  const customer = await stripe.customers.create();
+  const intent = await stripe.setupIntents.create({
+    customer: customer.id,
+    payment_method_types: ['card'],
+  });
+  if (intent) {
+    const user = await User.findByIdAndUpdate(req.user.id, { stripe_id: customer.id, stripe_intent_id: intent.id });
+    res.status(201).json({ user });
+  } else {
+    res.status(500);
+    throw new Error('unable to set up payment intent');
+  }
+});
+
+exports.getPublicKey = (req, res) => {
+  res.json({ public_key: process.env.STRIPE_PUBLIC_KEY });
+};
+
+exports.getSecret = asyncHandler(async (req, res, next) => {
+  const intent_id = await User.findById(req.user.id).then((user) => {
+    return user.stripe_intent_id;
+  });
+  const intent = await stripe.setupIntents.retrieve(intent_id);
+  if (intent) {
+    res.status(201).json({ client_secret: intent.client_secret });
```

Comment:
Please avoid using status code 201 as it is for resource creation :)

Ref: https://www.restapitutorial.com/httpstatuscodes.html